### DISCUSSION TITLE
Prevent Texts Over Writing each other

### DISCUSSION THREAD

**Discussion by naromwerd** [ImageMagick/ImageMagick#4350] [Help] [answered]
The following commands create a canvas and overlay some text using the -draw command.
Is the an easy way to prevent one text overwriting another?
Here is my example. Thanks in advance

magick convert -size 1280x1024 xc:none CANVAS.bmp
magick convert -font Arial-Bold -pointsize 12 -fill black -undercolor none -draw "text 1064,206 '1YI5A.SHUT DIPD01'" -undercolor none -draw "text 1042,206 '1YI5A.OPEN DIPD01'" -undercolor none -draw "text 802,214 '1TANK7.TXT TXTPD01'" -undercolor none -draw "text 802,307 '1TANK12.TXT TXTPD01'" -undercolor none -draw "text 802,276 '1TANK9.TXT TXTPD01'" -undercolor none -draw "text 802,459 '1PD1.TXT TXTPD01'" -undercolor none -draw "text 802,492 '1TANK1.TXT TXTPD01'" -undercolor none -draw "text 802,430 '1TANK20.TXT TXTPD01'" -undercolor none -draw "text 802,245 '1TANK5.TXT TXTPD01'" -undercolor none -draw "text 802,368 '1TANK6.TXT TXTPD01'" -undercolor none -draw "text 1109,138 '1TANK1.TEXT2 TXTPD01'" -undercolor none -draw "text 748,132 '1TANK1.TEXT TXTPD01'" -undercolor none -draw "text 914,707 '1DAT_TARE1 DATPD02'" -undercolor none -draw "text 565,430 '1DAT08 DATPD02'" -undercolor none -draw "text 565,461 '1DAT09 DATPD02'" -undercolor none -draw "text 565,492 '1DAT10 DATPD02'" -undercolor none -draw "text 565,369 '1DAT06 DATPD02'" -undercolor none -draw "text 565,399 '1DAT07 DATPD02'" -undercolor none -draw "text 565,245 '1DAT02 DATPD02'" -undercolor none -draw "text 565,276 '1DAT03 DATPD02'" -undercolor none -draw "text 565,307 '1DAT04 DATPD02'" -undercolor none -draw "text 565,214 '1DAT01 DATPD02'" -undercolor none -draw "text 679,727 'HEAD_B_ON DIPD01'" -undercolor none -draw "text 679,693 'HEAD_A_ON DIPD01'" -undercolor none -draw "text 299,640 'TARE_W_1 DATPD01'" -undercolor none -draw "text 961,732 '1D_COMP_1 DOPD01'" -undercolor none -draw "text 672,658 '1D_RESET1 DOPD01'" -undercolor none -draw "text 960,696 '1D_TARE1 DOPD01'" -undercolor none -draw "text 671,622 '1D_CLEAR1 DOPD01'" -undercolor none -draw "text 872,214 '1AC_W_01 AIPD01'" -undercolor none -draw "text 955,214 '1A_AB01 AOPD01'" -undercolor none -draw "text 300,671 '1SIC1.OUT AOPD01'" -undercolor none -draw "text 299,612 '1WIC101.IN AIPD01'" -undercolor none -draw "text 299,577 '1LI130.IN AIPD01'" -undercolor none -draw "text 977,206 '1D_AB01.OUT DOPD01'" -undercolor none -draw "text 62,394 '1MOT122 MOTPD01'" -undercolor none -draw "text 968,618 '1D_AB12.OUT DOPD01'" -undercolor none -draw "text 969,659 '1D_AB13.OUT DOPD01'" -undercolor none -draw "text 518,215 '1TEXT_AB01 TXTPD01'" -undercolor none -draw "text 518,246 '1TEXT_AB02 TXTPD01'" -undercolor none -draw "text 518,277 '1TEXT_AB03 TXTPD01'" -undercolor none -draw "text 518,307 '1TEXT_AB04 TXTPD01'" -undercolor none -draw "text 518,370 '1TEXT_AB06 TXTPD01'" -undercolor none -draw "text 518,401 '1TEXT_AB07 TXTPD01'" -undercolor none -draw "text 518,431 '1TEXT_AB08 TXTPD01'" -undercolor none -draw "text 518,462 '1TEXT_AB09 TXTPD01'" -undercolor none -draw "text 518,493 '1TEXT_AB10 TXTPD01'" -undercolor none -draw "text 872,245 '1AC_W_02 AIPD01'" -undercolor none -draw "text 955,245 '1A_AB02 AOPD01'" -undercolor none -draw "text 977,237 '1D_AB02.OUT DOPD01'" -undercolor none -draw "text 872,276 '1AC_W_03 AIPD01'" -undercolor none -draw "text 872,307 '1AC_W_04 AIPD01'" -undercolor none -draw "text 955,274 '1A_AB03 AOPD01'" -undercolor none -draw "text 955,307 '1A_AB04 AOPD01'" -undercolor none -draw "text 977,266 '1D_AB03.OUT DOPD01'" -undercolor none -draw "text 977,299 '1D_AB04.OUT DOPD01'" -undercolor none -draw "text 872,400 '1AC_W_07 AIPD01'" -undercolor none -draw "text 872,430 '1AC_W_08 AIPD01'" -undercolor none -draw "text 955,400 '1A_AB07 AOPD01'" -undercolor none -draw "text 955,430 '1A_AB08 AOPD01'" -undercolor none -draw "text 977,392 '1D_AB07.OUT DOPD01'" -undercolor none -draw "text 977,423 '1D_AB08.OUT DOPD01'" -undercolor none -draw "text 872,369 '1AC_W_06 AIPD01'" -undercolor none -draw "text 955,369 '1A_AB06 AOPD01'" -undercolor none -draw "text 977,361 '1D_AB06.OUT DOPD01'" -undercolor none -draw "text 872,461 '1AC_W_09 AIPD01'" -undercolor none -draw "text 872,491 '1AC_W_10 AIPD01'" -undercolor none -draw "text 955,461 '1A_AB09 AOPD01'" -undercolor none -draw "text 954,491 '1A_AB10 AOPD01'" -undercolor none -draw "text 977,454 '1D_AB09.OUT DOPD01'" -undercolor none -draw "text 977,484 '1D_AB10.OUT DOPD01'" -undercolor none -draw "text 316,740 '1WATER_BASE AOPD01'" -undercolor none -draw "text 264,512 '1TEXT_135 TXTPD01'" -undercolor none -draw "text 241,418 '1LI130.IN AIPD51'" -undercolor none -draw "text 333,183 '1TEXT_TANKFULL1 TXTPD01'" -undercolor none -draw "text 1042,237 '1YI12A.OPEN DIPD01'" -undercolor none -draw "text 1064,237 '1YI12A.SHUT DIPD01'" -undercolor none -draw "text 1042,266 '1YI1A.OPEN DIPD01'" -undercolor none -draw "text 1064,266 '1YI1A.SHUT DIPD01'" -undercolor none -draw "text 1042,299 '1YI2A.OPEN DIPD01'" -undercolor none -draw "text 1064,299 '1YI2A.SHUT DIPD01'" -undercolor none -draw "text 1042,360 '1YI9B.OPEN DIPD01'" -undercolor none -draw "text 1064,360 '1YI9B.SHUT DIPD01'" CANVAS.bmp CANVAS_DATA.bmp

**Comment by snibgo** [comment, marked as answer]
> Is the an easy way to prevent one text overwriting another?

Sadly, no.

**Reply by naromwerd** [reply]
Thank you
